handlers/oauth/oauth_test.go

```diff
@@ -0,0 +1,286 @@
+}
+
+func Test_OauthCallbackUserAuthError(t *testing.T) {
+	completeUserAuth = func(res http.ResponseWriter, req *http.Request) (goth.User, error) {
+		return goth.User{}, errors.New("uh oh something went bad")
+	}
+	rec := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(rec)
+	OauthCallback(ctx)
+	result := rec.Result()
+	if result.StatusCode != http.StatusInternalServerError {
+		t.Fatalf("Expected %d, got %d", http.StatusInternalServerError, result.StatusCode)
+	}
+	var responseJSON OauthErrorResponse
+	jsonError := json.Unmarshal(rec.Body.Bytes(), &responseJSON)
+	if jsonError != nil {
+		t.Fatalf(
+			"OauthCallback response expected OauthErrorResponse format, unmarshal failed with %s",
+			jsonError,
+		)
+	}
+}
+
+func Test_OauthCallbackUserFetchError(t *testing.T) {
+	initStore()
+	completeUserAuth = func(res http.ResponseWriter, req *http.Request) (goth.User, error) {
+		return goth.User{
+			UserID: "error",
+		}, nil
+	}
+	rec := httptest.NewRecorder()
+
+	ctx, _ := gin.CreateTestContext(rec)
+
+	newQuery := url.Values{
+		"provider": []string{"twitter"},
+	}
+	req, httpRequestErr := http.NewRequest("POST", "/", nil)
+	if httpRequestErr != nil {
+		t.Fatal(httpRequestErr)
+	}
+	req.URL.RawQuery = newQuery.Encode()
+	ctx.Request = req
+	OauthCallback(ctx)
+	result := rec.Result()
+	if result.StatusCode != http.StatusInternalServerError {
+		t.Fatalf("Expected %d, got %d", http.StatusInternalServerError, result.StatusCode)
+	}
+
+	var responseJSON OauthErrorResponse
+	jsonError := json.Unmarshal(rec.Body.Bytes(), &responseJSON)
+	if jsonError != nil {
+		t.Fatalf(
+			"OauthCallback response expected OauthErrorResponse format, unmarshal failed with %s",
+			jsonError,
+		)
+	}
+	if responseJSON.Error != "Issue finding user" {
+		t.Fatalf("Expected %s but recieved %s", "Issue finding user", responseJSON.Error)
+	}
+}
+
+func Test_OauthCallbackUserCreationError(t *testing.T) {
+	initStore()
+	completeUserAuth = func(res http.ResponseWriter, req *http.Request) (goth.User, error) {
+		return goth.User{
+			UserID: "0",
+		}, nil
+	}
+	rec := httptest.NewRecorder()
+
+	ctx, _ := gin.CreateTestContext(rec)
+
+	newQuery := url.Values{
+		"provider": []string{"twitter"},
+	}
+	req, httpRequestErr := http.NewRequest("POST", "/", nil)
+	if httpRequestErr != nil {
+		t.Fatal(httpRequestErr)
+	}
+	req.URL.RawQuery = newQuery.Encode()
+	ctx.Request = req
+	OauthCallback(ctx)
+	result := rec.Result()
+	if result.StatusCode != http.StatusOK {
+		t.Fatalf("Expected %d, got %d", http.StatusInternalServerError, result.StatusCode)
+	}
+
+	var responseJSON model.UserIdentifier
+	jsonError := json.Unmarshal(rec.Body.Bytes(), &responseJSON)
+	if jsonError != nil {
+		t.Fatalf(
+			"OauthCallback response expected UserIdentifier format, unmarshal failed with %s",
+			jsonError,
+		)
+	}
+}
+
+func Test_OauthCallbackReturnsExistingUser(t *testing.T) {
+	store := initStore()
+	expectedUser, createUserErr := store.CreateUser(model.User{
```

Comment:
nit: Usually in a scenario like this where the error is immediately checked, I'd prefer just sticking to the naming convention of `err`. This comment applies to a few other places as well.

Reply:
Done with https://github.com/leaderboardsgg/leaderboard-backend/pull/69/commits/672925b928f6e568330b0f0dba62f5406e071d46